app/controllers/admin/admin_users_controller.rb

```diff
@@ -239,6 +239,18 @@ def send_notification
     redirect_to :action => :notify
   end
 
+  def troubleshoot
+    @user = User.find_by_login(params[:id])
+    @user.fix_user_subscriptions
+    @user.reindex_user_works
+    @user.set_user_work_dates
+    @user.reindex_user_bookmarks
+    @user.create_log_item(options = { action: ArchiveConfig.ACTION_TROUBLESHOOT, admin_id: current_admin.id })
+    flash[:notice] = ts('User account checked.')
```

Comment:
Yay, looking good! The only thing is the flash message got overlooked -- it should be something like "User account troubleshooting complete." or "User account troubleshooting successful."

After that, I think we're ready to go! :tada: